sensor/common/compliance/node_inventory_handler_test.go

```diff
@@ -221,3 +221,62 @@ func (s *NodeInventoryHandlerTestSuite) TestInputChannelClosed() {
 	// The handler will stop as there are no more messages to handle
 	s.ErrorIs(h.Stopped().Wait(), errInputChanClosed)
 }
+
+func (s *NodeInventoryHandlerTestSuite) generateNilTestInputNoClose(numToProduce int) (chan *storage.NodeInventory, concurrency.StopperClient) {
+	input := make(chan *storage.NodeInventory)
+	st := concurrency.NewStopper()
+	go func() {
+		defer st.Flow().ReportStopped()
+		for i := 0; i < numToProduce; i++ {
+			select {
+			case <-st.Flow().StopRequested():
+				return
+			case input <- nil:
+			}
+		}
+	}()
+	return input, st.Client()
+}
+
+func (s *NodeInventoryHandlerTestSuite) TestHandlerNilInput() {
+	ch, producer := s.generateNilTestInputNoClose(10)
+	defer close(ch)
+	h := NewNodeInventoryHandler(ch, &mockAlwaysHitNodeIDMatcher{})
+	s.NoError(h.Start())
+	consumer := consumeAndCount(h.ResponsesC(), 0)
+	s.NoError(producer.Stopped().Wait())
+	s.NoError(consumer.Stopped().Wait())
+
+	h.Stop(nil)
+	s.NoError(h.Stopped().Wait())
+}
+
+func (s *NodeInventoryHandlerTestSuite) TestHandlerNodeUnknown() {
+	ch, producer := s.generateTestInputNoClose(10)
+	defer close(ch)
+	h := NewNodeInventoryHandler(ch, &mockNeverHitNodeIDMatcher{})
+	s.NoError(h.Start())
+	// expect consumer to get 0 messages - sensor should drop inventory when node is not found
+	consumer := consumeAndCount(h.ResponsesC(), 0)
+	s.NoError(producer.Stopped().Wait())
+	s.NoError(consumer.Stopped().Wait())
+
+	h.Stop(nil)
+	s.NoError(h.Stopped().Wait())
+}
+
+// mockAlwaysHitNodeIDMatcher always finds a node when GetNodeResource is called
+type mockAlwaysHitNodeIDMatcher struct{}
+
+// GetNodeID always finds a hardcoded ID "abc"
+func (c *mockAlwaysHitNodeIDMatcher) GetNodeID(nodename string) (string, error) {
+	return "abc", nil
+}
+
+// mockNeverHitNodeIDMatcher simulates inability to find a node when GetNodeResource is called
+type mockNeverHitNodeIDMatcher struct{}
+
+// GetNodeID neever finds a node and returns error hardcoded ID "abc"
```

Comment:
```suggestion
// GetNodeID never finds a node and returns error
```
Aside from the typo, a logic question: We return an error, no hardcoded ID at all, right?

Reply:
Yes, this simulates situation when the node is not found, so we return empty ID.